src/Microsoft.AspNetCore.Authentication/TokenExtensions.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Http.Features.Authentication;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public static class AuthenticationTokenExtensions
+    {
+        private static string TokenNamesKey = ".TokenNames";
+
+        public static void StoreTokens(this AuthenticationProperties properties, IEnumerable<AuthenticationToken> tokens)
+        {
+            if (properties == null)
+            {
+                throw new ArgumentNullException(nameof(properties));
+            }
+            if (tokens == null)
+            {
+                throw new ArgumentNullException(nameof(tokens));
+            }
+
+            var tokenNames = new List<string>();
+            foreach (var token in tokens)
+            {
+                // REVIEW: should probably check that there are no ; in the token name and throw or encode
+                tokenNames.Add(token.Name);
+                properties.Items[token.Name] = token.Value;
+            }
+            if (tokenNames.Count > 0)
+            {
+                properties.Items[TokenNamesKey] = string.Join(";", tokenNames.ToArray());
+            }
+        }
+
+        public static IEnumerable<AuthenticationToken> GetTokens(this AuthenticationProperties properties)
+        {
+            if (properties == null)
+            {
+                throw new ArgumentNullException(nameof(properties));
+            }
+
+            var tokens = new List<AuthenticationToken>();
+
+            if (properties.Items.ContainsKey(TokenNamesKey))
+            {
+                var tokenNames = properties.Items[TokenNamesKey].Split(';');
+                foreach (var name in tokenNames)
+                {
+                    if (properties.Items.ContainsKey(name))
+                    {
+                        tokens.Add(new AuthenticationToken { Name = name, Value = properties.Items[name] });
+                    }
+                }
+            }
+
+            return tokens;
+        }
+
+        public static Task<string> GetTokenAsync(this AuthenticationManager manager, string tokenName)
+        {
+            return manager.GetTokenAsync(AuthenticationManager.AutomaticScheme, tokenName);
+        }
+
+
+        public static async Task<string> GetTokenAsync(this AuthenticationManager manager, string signedInScheme, string tokenName)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+            if (signedInScheme == null)
+            {
+                throw new ArgumentNullException(nameof(signedInScheme));
+            }
+            if (tokenName == null)
```

Comment:
This is just a sugar/convention for callers to use, I don't think we should be opinionated on whether empty schemes or keys are valid or not, shrug.  My inclination is to leave them be as opposed to throwing.